api/include/opentelemetry/trace/attribute_value.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <cstdint>
+
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/variant.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+using AttributeValue = nostd::variant<bool,
+                                      int,
+                                      int64_t,
+                                      unsigned int,
+                                      uint64_t,
+                                      double,
+                                      nostd::string_view,
+                                      nostd::span<const bool>,
+                                      nostd::span<const int>,
+                                      nostd::span<const int64_t>,
+                                      nostd::span<const unsigned int>,
+                                      nostd::span<const uint64_t>,
+                                      nostd::span<const nostd::string_view>>;
```

Comment:
There's also this discussion going on: https://github.com/open-telemetry/opentelemetry-specification/issues/376. 

I'm not sure how serious that is, but it seems the idea is getting traction and is actively discussed currently. Maybe @reiley knows how likely this is to come and whether we should consider it.